WordPress/gutenberg
# Post Editor: Force iframe editor when zoom-out mode

Fixes #64259, #62943
Alternative to #64264

## What?

This PR forces the iframe editor when in zoom-out mode to ensure the zoom-out functionality works properly.

## Why?

Currently, the `useShouldIframe` hook will prevent the editor from being an iframe if all blocks are not all v3 or have meta boxes. If it is not an iframe, the new view "Desktop (50%)" and the zoom-out mode for experimental pattern inserters will not work.

## How?

Update the `useShouldIframe` hook to force the iframe editor when in zoom-out mode.

## Testing Instructions

- Enable "Enable zoomed out view when patterns are browsed in the inserter" from the experimental page beforehand.
- Enable custom fields in the post editor. This will make the editor not an iframe.
- If you select "Desktpp (50%)" from the view options, the editor will be an iframe, and zoom out should work.
- If you select the pattern category from the main inserter, the editor will be an iframe, and zoom out should work.

## Screenshots or screencast 

https://github.com/user-attachments/assets/451c3679-fcf0-492f-be9c-ebfd3ea7ef9a
